pkg/capabilities/consensus/ocr3/requests/store.go

```diff
@@ -30,16 +30,16 @@ func (s *Store) GetByIDs(ctx context.Context, requestIDs []string) []*Request {
 	for _, r := range requestIDs {
 		gr, ok := s.requests[r]
 		if ok {
-			o = append(o, gr)
+			o = append(o, gr.Copy())
```

Comment:
i find this unintuitive and likely to lead to subtle bugs if more hands are in this code base.

how can you make it more obvious when Copy is needed?

at minimum a short comment around at the usage sites and func comment. 

a defensive alternative might be to type def something like  
```
type copiedRequests []*request
 func (c *copiedRequests) append(r *request) {
 c = append(c, r.Copy)
 }
 ```
 
 something like that leverage the compiler to help defend intermixing and confusing append(,r) vs append(, r.Copy())

Reply:
I added some documentation -- I preferred this to something more defensive which feels odd to me (more complicated and I don't feel like it would reduce bugs; new methods would explicitly need to opt in to using this)